main.py

```diff
@@ -115,26 +115,30 @@ def process(data, factors):
             average[factor_set][surface] = np.average(np.array(average[factor_set][surface]), axis=0)
     return average
 
-def sort_directions(data):
+def choose_direction(data, average):
     user_surfaces = []
     for direction_name, surface_arrays in data.items():
-        user_surfaces.append((direction_name, len(surface_arrays.keys()),surface_arrays.keys()))
-    return sorted(user_surfaces, key=lambda x: x[1], reverse=True)
-
+        user_surfaces.append((direction_name, len(surface_arrays.keys())))
+    #get the number of surfaces stored in average as keys
+    user_surfaces = sorted(user_surfaces, key=lambda x: x[1], reverse=True)
+    chosen_surface_num = user_surfaces[0][1]
+    average_surface_num = len(average[next(iter(average))].keys())
+    #I am not sure if this is really necessary to check if they are equal
+    if average_surface_num == chosen_surface_num:
+        return user_surfaces[0][0]
```

Comment:
If we're returning just one final chosen surface from ```user_surfaces```, wouldn't it be better if we eliminate ```user_surfaces``` and make a variable only storing the chosen surface?

Reply:
let me think think how to do it sin...

Reply:
```user_surfaces``` is removed in 46d9a6ba1b1df05f123a3ca0479fb4e7e9080320, resolving discussion.